xbmc/addons/binary/interfaces/api2/AddonInterfaceBase.cpp

```diff
@@ -0,0 +1,197 @@
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "AddonInterfaceBase.h"
+
+#include "Addon/Addon_Audio.h"
+#include "Addon/Addon_Directory.h"
+#include "Addon/Addon_File.h"
+#include "Addon/Addon_General.h"
+#include "Addon/Addon_Network.h"
+#include "AudioEngine/Addon_AudioEngineGeneral.h"
+#include "AudioEngine/Addon_AudioEngineStream.h"
+#include "GUI/Addon_GUIGeneral.h"
+#include "GUI/Addon_GUIControlButton.h"
+#include "GUI/Addon_GUIControlEdit.h"
+#include "GUI/Addon_GUIControlFadeLabel.h"
+#include "GUI/Addon_GUIControlImage.h"
+#include "GUI/Addon_GUIControlLabel.h"
+#include "GUI/Addon_GUIControlProgress.h"
+#include "GUI/Addon_GUIControlRadioButton.h"
+#include "GUI/Addon_GUIControlRendering.h"
+#include "GUI/Addon_GUIControlSettingsSlider.h"
+#include "GUI/Addon_GUIControlSlider.h"
+#include "GUI/Addon_GUIControlSpin.h"
+#include "GUI/Addon_GUIControlTextBox.h"
+#include "GUI/Addon_GUIDialogExtendedProgressBar.h"
+#include "GUI/Addon_GUIDialogFileBrowser.h"
+#include "GUI/Addon_GUIDialogKeyboard.h"
+#include "GUI/Addon_GUIDialogNumeric.h"
+#include "GUI/Addon_GUIDialogOK.h"
+#include "GUI/Addon_GUIDialogProgress.h"
+#include "GUI/Addon_GUIDialogSelect.h"
+#include "GUI/Addon_GUIDialogTextViewer.h"
+#include "GUI/Addon_GUIDialogYesNo.h"
+#include "GUI/Addon_GUIListItem.h"
+#include "GUI/Addon_GUIWindow.h"
+#include "InputStream/Addon_InputStream.h"
+#include "PVR/Addon_PVR.h"
+#include "Peripheral/Addon_Peripheral.h"
+#include "Player/Addon_InfoTagMusic.h"
+#include "Player/Addon_InfoTagVideo.h"
+#include "Player/Addon_PlayList.h"
+#include "Player/Addon_Player.h"
+
+#include "Application.h"
+#include "addons/Addon.h"
+#include "addons/binary/ExceptionHandling.h"
+#include "addons/binary/interfaces/AddonInterfaces.h"
+#include "addons/kodi-addon-dev-kit/include/kodi/api2/.internal/AddonLib_internal.hpp"
+#include "addons/kodi-addon-dev-kit/src/api2/version.h"
+#include "utils/log.h"
+
+using namespace ADDON;
+
+namespace V2
+{
+namespace KodiAPI
+{
+extern "C"
+{
+
+CAddonInterfaceAddon::CAddonInterfaceAddon(CAddon* addon)
+  : ADDON::IAddonInterface(addon, ADDON_API_LEVEL, ADDON_API_VERSION),
+    m_callbacks(new CB_AddOnLib)
+{
+  m_callbacks->addon_log_msg = addon_log_msg;
+  m_callbacks->free_string   = free_string;
+
+  AddOn::CAddOnGeneral                  ::Init(m_callbacks);
```

Comment:
You mean to change to `AddOn::CAddOnGeneral::Init(m_callbacks);` without spaces? No problem an be changed.